src/rgw/rgw_process.cc

```diff
@@ -79,6 +79,78 @@ int rgw_process_authenticated(RGWHandler_REST * const handler,
     return ret;
   }
 
+  /* Check if OPA is used to authorize requests */
+  if (s->cct->_conf->rgw_use_opa_authz) {
```

Comment:
this looks like the right place to put it. but i'd prefer to move the rest into a helper function in its own source file so it's not all inline here. something like `rgw_opa_authorize()` or `rgw::opa::authorize()` in rgw_opa.h/cc?

Reply:
@cbodley @mattbenjamin  After adding rgw_opa.h/cc files, what is correct way to build radosgw ? `make radosgw` gives `undefined reference` errors which make sense. Can you please point me to any doc or other resource that can help ? Thanks

Reply:
in src/rgw/CMakeLists.txt, you'll need to add `rgw_opa.cc` to the list of `rgw_a_srcs`

Reply:
Thanks @cbodley !  @mattbenjamin @cbodley I have moved the OPA code into a separate file so that it's not inline in `rgw_process.cc`